lib/datasource/github-releases/index.ts

```diff
@@ -1,22 +1,186 @@
+import hasha from 'hasha';
 import { logger } from '../../logger';
 import * as packageCache from '../../util/cache/package';
+import { GithubHttp } from '../../util/http/github';
+import { newlineRegex, regEx } from '../../util/regex';
+import { ensureTrailingSlash } from '../../util/url';
 import type { DigestConfig, GetReleasesConfig, ReleaseResult } from '../types';
-import {
-  cacheNamespace,
-  getApiBaseUrl,
-  getGithubRelease,
-  getSourceUrl,
-  http,
-  id,
-} from './common';
-import { findDigestAsset, mapDigestAssetToRelease } from './digest';
-import type { GithubRelease } from './types';
-
-export { id };
+import type { DigestAsset, GithubRelease, GithubReleaseAsset } from './types';
+
 export const customRegistrySupport = true;
 export const defaultRegistryUrls = ['https://github.com'];
 export const registryStrategy = 'first';
 
+const defaultSourceUrlBase = 'https://github.com/';
+export const id = 'github-releases';
+
+export const cacheNamespace = 'datasource-github-releases';
+export const http = new GithubHttp(id);
+
+async function findDigestFile(
+  release: GithubRelease,
+  digest: string
+): Promise<DigestAsset | null> {
+  const smallAssets = release.assets.filter(
+    (a: GithubReleaseAsset) => a.size < 5 * 1024
+  );
+  for (const asset of smallAssets) {
+    const res = await http.get(asset.browser_download_url);
+    for (const line of res.body.split(newlineRegex)) {
+      const [lineDigest, lineFn] = line.split(regEx(/\s+/), 2);
+      if (lineDigest === digest) {
+        return {
+          assetName: asset.name,
+          digestedFileName: lineFn,
+          currentVersion: release.tag_name,
+          currentDigest: lineDigest,
+        };
+      }
+    }
+  }
+  return null;
+}
+
+function inferHashAlg(digest: string): string {
```

Comment:
Can't we leave those small functions in the common file? So this file don't get so big?

Reply:
Let's do this in the next phase, after migration to class is done

Reply:
Why not simply revert this PR changes partially instead, so don't need to change it again?

Reply:
It seems like the initial set of function considered as *util* ones may be different after switching to class-based implementation. I would like to deal with this later, as the separate step.

Reply:
Ok